### ISSUE TITLE
Nested Suspense not working

### ISSUE THREAD

**Issue by dmmikkel** [honojs/hono#1658]
### What version of Hono are you using?

next

### What runtime/platform is your app running on?

Deno

### What steps can reproduce the bug?

Set up request and components:

```jsx
const AsyncThing = async () => {
  console.log("Async start");
  await new Promise((resolve) => setTimeout(resolve, 1000));
  console.log("Async end");
  return <div class="bg-red-500">Async thing</div>;
};

const Home = async () => {
  console.log("Home");
  await new Promise((resolve) => setTimeout(resolve, 1000));

  return (
    <div>
      <h1>Home</h1>
      <Suspense fallback={<div>Loading AsyncThing...</div>}>
        <AsyncThing />
      </Suspense>
    </div>
  );
};

const TestPage: FC = () => {
  return (
    <div>
      <Suspense fallback={<div>Loading Home...</div>}>
        <Home />
      </Suspense>
    </div>
  );
};

app.get('/', (c) => {
    console.log("------------------")
    console.log("Request start")

    return c.stream(async s => {
        s.write("<!DOCTYPE html>");
        const stream = renderToReadableStream(<TestPage />);
        for await (const chunk of stream) {
            s.write(chunk);
        }
        s.close();
    }, {
        headers: {
            'Content-Type': 'text/html; charset=utf-8',
        },
        status: 200,
    })
});
```

Then visit "/".

### What is the expected behavior?

First the outer suspense to resolve and stream down it's result, then the inner suspense should do the same. Taking in total 2s.

### What do you see instead?

Only the outer suspense is resolved and renders it's result. The request ends after 1s with "Loading AsyncThing..." visible.

Server side I can observe the AsyncThing component actually invoked, but the result is not sent to user as the request already ended.

**Comment by usualoma** [comment]
Thanks again!
This issue will be fixed in #1660.

**Comment by dmmikkel** [comment]
Thanks for the fix :)

**Comment by yusukebe** [comment]
Fixed with #1660.
